treeherder/changelog/models.py

```diff
@@ -0,0 +1,32 @@
+from django.db import models
+
+
+class Changelog(models.Model):
+    id = models.CharField(max_length=255, primary_key=True)
+    date = models.DateTimeField(db_index=True)
+    author = models.CharField(max_length=100)
+    owner = models.CharField(max_length=100)
+    project = models.CharField(max_length=100)
+    project_url = models.CharField(max_length=360)
+    message = models.CharField(max_length=360)
+    description = models.CharField(max_length=360)
+    type = models.CharField(max_length=100)
+    url = models.CharField(max_length=360)
+
+    class Meta:
+        db_table = 'changelog_entry'
+
+    def __str__(self):
+        return "[%s] %s by %s" % (self.id, self.message, self.author)
+
+
+class ChangeLogFile(models.Model):
```

Comment:
Could there be multiple changelog files associated to a changelog entry? Wouldn't a one-to-one relation suffice?

Reply:
one changeset contains **several** files, so it's a one-to-many relation. now we could also use a many-to-many relation since a given file could be part of several changesets as well, but given the volume of the data I don't think it really matters. Can't be one-to-one though.